--- leksah-0.14.4.0/leksah.cabal~	2015-06-13 22:38:15.316736725 +0900
+++ leksah-0.14.4.0/leksah.cabal	2015-06-13 22:39:21.572892050 +0900
@@ -238,7 +238,7 @@
       build-depends: webkitgtk3 -any, webkitgtk3-javascriptcore -any
     else
       build-depends: webkit -any, webkit-javascriptcore -any
-    build-depends: jsaddle -any, ghcjs-codemirror -any, hamlet -any, blaze-html -any,
+    build-depends: jsaddle -any, ghcjs-codemirror -any, blaze-html -any,
                    lens -any, pretty-show >=1.6.3 && <1.7
     cpp-options: -DWEBKITGTK -DLEKSAH_WITH_CODE_MIRROR
 
