--- leksah-0.13.4.3/leksah.cabal~	2014-09-03 13:41:33.867570388 +0900
+++ leksah-0.13.4.3/leksah.cabal	2014-09-03 13:46:33.761363375 +0900
@@ -306,7 +306,11 @@
         ghc-options: -threaded
 
     hs-source-dirs: main
-    build-depends: leksah == 0.13.4.3, base >= 4.0.0.0 && <=4.8, gtk3 -any
+    build-depends: leksah == 0.13.4.3, base >= 4.0.0.0 && <=4.8
+    if flag(gtk3)
+        build-depends: gtk3 >=0.12.4 && <0.13
+    else
+        build-depends: gtk >=0.12.4 && <0.13
     if os(linux) && flag(loc)
         cpp-options: -DLOCALIZATION
         build-depends: hgettext, setlocale
